src/Servers/Kestrel/Core/src/Internal/Http3/Http3Connection.cs

```diff
@@ -133,6 +133,11 @@ public void StopProcessingNextRequest(bool serverInitiated)
 
                 if (Interlocked.CompareExchange(ref _gracefulCloseInitiator, initiator, GracefulCloseInitiator.None) == GracefulCloseInitiator.None)
                 {
+                    // https://quicwg.org/base-drafts/draft-ietf-quic-http.html#section-5.2-11
+                    // An endpoint that completes a graceful shutdown SHOULD use the H3_NO_ERROR error code
+                    // when closing the connection.
+                    _errorCodeFeature.Error = (long)Http3ErrorCode.NoError;
```

Comment:
Should we just make NoError a long?

Reply:
What is your reasonsing here? Is it about the cast?

Http3ErrorCode is an enum. I believe you always need to cast when going from an enum to a number.

Reply:
Yeah nvrm it is already a long.